source/application/DbgHelp/DbgHelpStackWalkCallback.cpp

```diff
@@ -69,109 +69,252 @@ const std::wstring DbgHelpStackWalkCallback::GetFunctionName()
+        parameters.emplace_back(param_info.m_type, param_info.m_name, GetParameterValue(param_info));
     }
 
-    return m_param_info.size();
+    return parameters;
 }
 
-DbgHelpArgType DbgHelpStackWalkCallback::GetParameterType(DWORD num)
+CComPtr<IDiaSymbol> DbgHelpStackWalkCallback::GetFunctionSymbol()
 {
-    return DbgHelpArgType();
+    CComPtr<IDiaSymbol> symbol;
+    HRESULT rv = m_mod_info->m_module_symbol_session->findSymbolByVA(GetProgramCounter(), SymTagFunction, &symbol);
+    return symbol;
 }
 
-std::wstring DbgHelpStackWalkCallback::GetParameterTypeName(DWORD num)
+ULONGLONG DbgHelpStackWalkCallback::GetProgramCounter()
 {
-    return std::wstring();
+    ULONGLONG pc;
+    m_frame->get_registerValue(CV_REG_EIP, &pc);
+    return pc;
 }
 
-std::wstring DbgHelpStackWalkCallback::GetParameterName(DWORD num)
+// In these functions length MUST match the "expected" size of the basic types,
+// otherwise they will be read incorrectly.
+static std::optional<DbgHelpBasicType::BasicType> GetDbgHelpIntType(ULONGLONG length, bool is_signed)
 {
-    return std::wstring();
+    switch (length)
+    {
+    case 1:
+        return is_signed ? DbgHelpBasicType::BasicType::Int8 : DbgHelpBasicType::BasicType::UnsignedInt8;
+    case 2:
+        return is_signed ? DbgHelpBasicType::BasicType::Int16 : DbgHelpBasicType::BasicType::UnsignedInt16;
+    case 4:
+        return is_signed ? DbgHelpBasicType::BasicType::Int32 : DbgHelpBasicType::BasicType::UnsignedInt32;
+    case 8:
+        return is_signed ? DbgHelpBasicType::BasicType::Int64 : DbgHelpBasicType::BasicType::UnsignedInt64;
+    default:
+        return std::nullopt;
+    }
 }
 
-std::shared_ptr<void> DbgHelpStackWalkCallback::GetParameterValue(DWORD num)
+static std::optional<DbgHelpBasicType::BasicType> GetDbgHelpFloatType(ULONGLONG length)
 {
-    return std::shared_ptr<void>();
+    switch (length)
+    {
+    case 4:
+        return DbgHelpBasicType::BasicType::Float;
+    case 8:
+        return DbgHelpBasicType::BasicType::Double;
+    default:
+        return std::nullopt;
+    }
 }
 
-CComPtr<IDiaSymbol> DbgHelpStackWalkCallback::GetFunctionSymbol()
+static std::optional<DbgHelpBasicType> GetDbgHelpBasicType(ULONGLONG length, DWORD basic_type)
 {
-    CComPtr<IDiaSymbol> symbol;
-    HRESULT rv = m_mod_info->m_module_symbol_session->findSymbolByVA(GetProgramCounter(), SymTagFunction, &symbol);
-    return symbol;
+    switch (basic_type)
+    {
+    case btChar:
+        return DbgHelpBasicType(DbgHelpBasicType::BasicType::Char);
+    case btInt:
+    case btLong:
+        {
+            auto int_type = GetDbgHelpIntType(length, true);
+            if (int_type.has_value())
+            {
+                return DbgHelpBasicType(int_type.value());
+            }
+        }
+        break;
+    // Treat wchar_t, char16_t and char32_t as integers of the given size as they don't have a defined size.
+    case btWChar:
+    case btChar16:
+    case btChar32:

```

Comment:
Yes, however it's not very clear how if, for instance, the target binary was compiled with a compiler which has 4-byte `wchar_t`, and this binary used that as an argument to a function, and passed some high UTF-8 code point. In this case it won't fit into MSVC's 2-byte `wchar_t`. I guess I can check if those types are the "expected" size and print them then, otherwise do the current handling.